tap_github/streams.py

```diff
@@ -158,6 +158,113 @@ def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
     ).to_dict()
 
 
+class EventsStream(GitHubStream):
+    """
+    Defines 'Events' stream.
+    Issue events are fetched from the repository level (as opposed to per issue)
+    to optimize for API quota usage.
+    """
+
+    name = "events"
+    path = "/repos/{org}/{repo}/events"
+    primary_keys = ["id"]
+    replication_key = "created_at"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys = ["repo", "org"]
+    ignore_parent_replication_key = False
+
+    def get_records(self, context: Optional[dict] = None) -> Iterable[Dict[str, Any]]:
+        """Return a generator of row-type dictionary objects.
+
+        Each row emitted should be a dictionary of property names to their values.
+        """
+        if context and context.get("events", None) == 0:
+            self.logger.debug(f"No events detected. Skipping '{self.name}' sync.")
+            return []
+
+        return super().get_records(context)
+
+    def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
+        # TODO - We should think about the best approach to handle this. An alternative would be to do
+        # a 'dumb' tap that just keeps the same schemas as GitHub without renaming these objects to "target_".
```

Comment:
why worry about renaming the fields to target_? Since this is the extract step I'm thinking it's fine to just keep the fields as is. (We are trying to match the original schema as much as possible I believe)

Reply:
I would agree, but the way the tap is created, it imports by default "repo" and "org" as the pair we are currently fetching information for. In some cases however, this will mean that the one we would add and the ones we get from GitHub have different value. I don't have a strong preference but I thought I would flag that this is a risk and let the upstream team decide what they prefer to do.

Reply:
Sounds good, let's highlight this when we make the the upstream PR